lib/macho/macho_file.rb

```diff
@@ -189,7 +189,38 @@ def add_command(lc)
       set_sizeofcmds(new_sizeofcmds)
       @raw_data[header.class.bytesize + sizeofcmds, cmd_raw.bytesize] = cmd_raw
 
-      # synchronize fields with the raw data
+      synchronize_fields! if synchronize
+    end
+
+    # Delete a load command from the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being deleted
+    # @param synchronize [Boolean] whether or not to synchronize the instance fields
+    # @return [void]
+    # @note This is public, but methods like {#delete_rpath} should be preferred.
+    #  Setting `synchronize` to false **will break the instance** unless
+    #  {#synchronize_fields!} is called **immediately** afterwards.
+    def delete_command(lc, synchronize = true)
+      @raw_data.slice!(lc.view.offset, lc.cmdsize)
+
+      # decrease the number of load commands by the number deleted
+      set_ncmds(ncmds - 1)
+
+      # decrease the total size of commands by the size of the deleted commands
+      set_sizeofcmds(sizeofcmds - lc.cmdsize)
+
+      # pad the space after the load commands to preserve offsets
+      null_pad = "\x00" * lc.cmdsize
+      @raw_data.insert(header.class.bytesize + sizeofcmds, null_pad)
+
+      synchronize_fields! if synchronize
+    end
+
+    # Synchronize the instance's fields with the raw Mach-O data.
+    # @return [void]
+    # @note This method is public, but should (almost) never need to be called.
+    #  The exception to this rule is when methods like {#add_command} and
+    #  #{delete_command} have been called with `synchronization_false`.
+    def synchronize_fields!
```

Comment:
I think we can drop the bang (`!`) at the end and I'm afraid this method name could be confusing. When I'm thinking of synchronization, I'm thinking more about flushing the changes that are only in memory to disk (or in this case to raw data), i.e. exactly in the opposite direction.

How do you feel about naming this `reload_fields`, `reload_from_raw_data` (a bit uglier but also clearer), or something similar (better ideas for a name are more than welcome)?